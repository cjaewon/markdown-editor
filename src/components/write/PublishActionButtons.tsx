import * as React from 'react';
import styled from 'styled-components';
import Button from '../common/Button';
import media from '../../lib/styles/media';

const PublishActionButtonsBlock = styled.div`
  display: flex;
  justify-content: flex-end;
  ${media.custom(767)} {
    margin-top: 2rem;
  }
`;

export interface PublishActionButtonsProps {
  onCancel: () => void;
  onPublish: () => void;
  edit: boolean;
}

const PublishActionButtons: React.FC<PublishActionButtonsProps> = ({
  onCancel,
  onPublish,
  edit,
}) => {
  return (
    <PublishActionButtonsBlock>
      <Button
        size="large"
        data-testid="cancelPublish"
        color="gray"
        onClick={onCancel}
      >
        취소
      </Button>
      <Button size="large" data-testid="publish" onClick={onPublish}>
        {edit ? '수정하기' : '출간하기'}
      </Button>
    </PublishActionButtonsBlock>
  );
};

export default PublishActionButtons;
