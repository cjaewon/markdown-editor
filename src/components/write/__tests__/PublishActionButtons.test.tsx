import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import PublishActionButtons, {
  PublishActionButtonsProps,
} from '../PublishActionButtons';

describe('PublishActionButtons', () => {
  const setup = (props: Partial<PublishActionButtonsProps> = {}) => {
    const initialProps: PublishActionButtonsProps = {
      onCancel: () => {},
      onPublish: () => {},
      edit: false,
    };
    const utils = render(<PublishActionButtons {...initialProps} {...props} />);
    const buttons = {
      cancel: utils.getByText('취소'),
      publish: utils.getByText(/(출간|수정)하기/),
    };
    return {
      ...utils,
      buttons,
    };
  };
  it('renders properly', () => {
    setup();
  });
  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('calls onCancel and onPublish', () => {
    const onCancel = jest.fn();
    const onPublish = jest.fn();
    const utils = setup({
      onCancel,
      onPublish,
    });
    fireEvent.click(utils.buttons.publish);
    expect(onPublish).toBeCalled();
    fireEvent.click(utils.buttons.cancel);
    expect(onCancel).toBeCalled();
  });
  it('changes button text when edit is true', () => {
    const { getByText } = setup({
      edit: true,
    });
    getByText('수정하기');
  });
});
