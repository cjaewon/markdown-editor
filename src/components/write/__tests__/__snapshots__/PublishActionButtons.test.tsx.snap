// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublishActionButtons matches snapshot 1`] = `
<div>
  <div
    class="sc-bwzfXH hlaomC"
  >
    <button
      class="sc-bdVaJa gBmPlh"
      color="gray"
      data-testid="cancelPublish"
    >
      취소
    </button>
    <button
      class="sc-bdVaJa dwcrM"
      color="teal"
      data-testid="publish"
    >
      출간하기
    </button>
  </div>
</div>
`;
